.github/workflows/upload.yml

```diff
@@ -8,38 +8,85 @@ on:
       - "src/technologies/*.json"
       - "src/categories.json"
       - "src/groups.json"
+      - "src/images/icons/**"
   workflow_dispatch:
+    inputs:
+      force_update:
+        description: 'Force update regardless of changes'
+        required: false
+        default: 'false'
 
 jobs:
   test:
     name: Test and upload to GCP
     runs-on: ubuntu-latest
     steps:
+
       - name: Checkout
         uses: actions/checkout@v4
         with:
           ref: ${{ github.event.pull_request.head.sha }}
-          fetch-depth: 0
 
       - name: Install dependencies
         run: yarn install
 
-      - name: Validate
-        run: yarn run validate
+      - name: Validate and Lint
+        run: |
+          yarn run lint
+          yarn run validate
 
-      - name: Run WebPageTest with unit tests
-        id: unit-test
-        env:
-          WPT_SERVER: "webpagetest.httparchive.org"
-          WPT_API_KEY: ${{ secrets.HA_API_KEY }}
-          PR_NUMBER: ${{ github.event.pull_request.number }}
-        run: yarn run test
+      - name: Get changed technology files
+        run: |
+          git log -1 HEAD^
+          git log -1 HEAD
 
-      - name: Upload to GCP
-        id: upload
+          git fetch origin main --depth=1
+          TECH_FILES_CHANGED=$(git diff --name-only HEAD^ HEAD -- 'src/technologies/*.json' 'src/categories.json' 'src/groups.json' | wc -l)
+          echo $TECH_FILES_CHANGED
+          echo "TECH_FILES_CHANGED=${TECH_FILES_CHANGED}" >> $GITHUB_ENV
+
+      - name: Upload technology rules to BigQuery
+        if: env.TECH_FILES_CHANGED != '' || inputs.force_update == 'true'
         env:
-          GCP_PROJECT_ID: ${{ secrets.GCP_PROJECT_ID }}
+          GCP_PROJECT_ID: 'httparchive'
```

Comment:
Not that it's that secret, but any reason not to use the secret here?

Reply:
I couldn't find arguments to use it.

Reply:
I can actually remove it completely and we use the parent project from service account.